### DISCUSSION TITLE
422 Unprocessable Entity error uploading attachment

### DISCUSSION THREAD

**Discussion by kswygert** [mozts2005/ZendeskApi_v2#552] [Q&A] [answered]
Using the latest .Net Zendesk API version 3.12.0, I'm getting the following exception. 

"The remote server returned an error: (422) Unprocessable Entity.Error content: {"error":"AttachmentUnprocessable","description":Attachment file could not be processed."}
Resource String: uploads.json?filename=C:\file.pdf +
File Name: C:\file.pdf
File Length: No Data

I'm passing in a pdf file as an "attachment".  It's failing on the UploadAttachment, but I'm passing in the file as a ZenFile.  Here's the code portion that is failing.

ZendeskApi api = new ZendeskApi("https://domain.zendesk.com/api/v2", "user", "password");

                    ZenFile file = new ZenFile
                    {
                        FileName = attachment                        
                    };
                    api.Attachments.UploadAttachment(file);

 

What am I missing to get this error?

**Comment by mozts2005** [comment, marked as answer]
You are missing a few things like the data and the file name should be just the name you want in zendesk not the path. 

https://github.com/Speedygeek/ZendeskApi_v2/blob/7b9aa5afb8c7860c8947a55c7a7ce6753d76efb7/tests/ZendeskApi_v2.Tests/AttachmentTests.cs#L16-L26

**Reply by kswygert** [reply]
Thank you for pointing me in the right direction.  I was able to resolve the issue using the Test example you provided.
